dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/sync/DataValuesSynchronization.java

```diff
@@ -0,0 +1,243 @@
+
+            final Date startTime = new Date();
+            final Date lastSuccessTime = getLastDataSynchSuccessFallback();
+            final int objectsToSync = dataValueService.getDataValueCountLastUpdatedAfter( lastSuccessTime, true );
+
+            if ( objectsToSync == 0 )
+            {
+                log.debug( "Skipping synchronization, no new or updated DataValues" );
+                return SynchronizationResult.newSuccessResultWithMessage( "Skipping synchronization, no new or updated DataValues" );
+            }
+
+            log.info( "Values: " + objectsToSync + " since last synchronization success: " + lastSuccessTime );
+
+            final String syncUrl = systemSettingManager.getSystemSetting( SettingKey.REMOTE_INSTANCE_URL ) + SyncEndpoint.DATA_VALUE_SETS.getPath();
+            final String username = (String) systemSettingManager.getSystemSetting( SettingKey.REMOTE_INSTANCE_USERNAME );
+            final String password = (String) systemSettingManager.getSystemSetting( SettingKey.REMOTE_INSTANCE_PASSWORD );
+
+            final SystemInstance instance = new SystemInstance( syncUrl, username, password );
+
+            log.info( "Remote server URL for DataValues POST sync: " + instance.getUrl() );
+
+            final int syncPageSize = (int) systemSettingManager.getSystemSetting( SettingKey.DATA_VALUES_SYNC_PAGE_SIZE );
+            final int pages = (objectsToSync / syncPageSize) + ((objectsToSync % syncPageSize == 0) ? 0 : 1);  //Have to use this as (int) Match.ceil doesn't work until I am casting int to doublee to use this as (int) Match.ceil doesn't work until I am casting int to double
+
+            log.info( "DataValues sync job has " + pages + " pages to sync. With page size: " + syncPageSize );
+
+            boolean syncResult = true;
+
+            for ( int i = 1; i <= pages; i++ )
+            {
+                log.info( String.format( "Syncing page %d, page size is: %d", i, syncPageSize ) );
+
+                if ( !sendDataValuesSyncRequest( instance, lastSuccessTime, syncPageSize, i, SyncEndpoint.DATA_VALUE_SETS ) )
+                {
+                    syncResult = false;
+                }
+                //TODO: Possible improvement is to update lastSuccessTime timestamp to those that were just synced. But it will require more effort
+            }
+
+            if ( syncResult )
+            {
+                long syncDuration = System.currentTimeMillis() - startTime.getTime();
+                log.info( "SUCCESS! DataValues synchronization was successfully done! It took " + syncDuration + " ms." );
+                setLastDataValuesSynchronizationSuccess( startTime );
+                return SynchronizationResult.newSuccessResultWithMessage( "DataValues synchronization done. It took " + syncDuration + " ms." );
+            }
+
+            return SynchronizationResult.newFailureResultWithMessage( "DataValues synchronization failed." );
+        }
+        catch ( Exception e )
```

Comment:
Is it possible to catch a more specific Exception?

Also if you are logging like you do, it should be more informative than "Exception: e"; We already know that :) Maybe "Failed to do <something>: e" That way we know what was going on. Maybe even log data if relevant, ie. "Failed to do <something>, <configuration used>"